app/controllers/api/v1x0/requests_controller.rb

```diff
@@ -22,14 +22,48 @@ def index
                  Request.includes(:stages)
                end
 
-        collection(reqs)
+        RBAC::Access.enabled? ? collection(rbac_scope(reqs)) : collection(reqs)
       end
 
       private
 
       def request_params
         params.permit(:name, :description, :requester_name, :content => {})
       end
+
+      def rbac_scope(relation)
+        access_obj = RBAC::Access.new('requests', 'read').process
+        raise Exceptions::NotAuthorizedError, "Not Authorized to list requests" unless access_obj.accessible?
+
+        # return error for using wrong path
+        raise Exceptions::NotAuthorizedError, "Current role cannot access #{request.path}" unless right_path?(access_obj)
+
+        return relation if access_obj.admin?
+
+        approver_id_list = access_obj.approver_id_list
```

Comment:
Can't we rewrite the following logic this way?
```
relation =
  if approver_endpoint?
    relation.where(:id => access_obj.approval_id_list)
  else
    relation.where(:id => access_obj.owner_id_list)
  end
Rails.logger.info(...)
relation
```
I don't see the case for double roles